jopa-impl/src/main/java/cz/cvut/kbss/jopa/model/metamodel/EntityLifecycleListenerManager.java

```diff
@@ -83,8 +83,8 @@ private void invokeCallbacks(Object instance, LifecycleEvent lifecycleEvent) {
     }
 
     private void invokeEntityListenerCallbacks(Object instance, LifecycleEvent lifecycleEvent) {
-        if (parent != null) {
-            parent.invokeEntityListenerCallbacks(instance, lifecycleEvent);
+        if (!parents.isEmpty()) {
```

Comment:
No need to check for emptiness before calling foreach.

Reply:
SOLVED - removed checks